### PR TITLE
[Core][Array1D] Improve initialization

### PR DESCRIPTION
**Pull request by philbucher**
**Description**
Quite often we need to initialize the contents of the array to zero before we can use them. This is cumbersome and tends to be forgotten which has often caused problems with uninitialized variables.

Hence this PR:
- default initializes the `array_1d`
- adds a ctor supporting [`std::initializer_list`](https://en.cppreference.com/w/cpp/utility/initializer_list) => with this we can finally do e.g. `array_1d<double, 3> arr {1.1, -2.3, 3.4};`

### CODE REVIEW

File: kratos/containers/array_1d.h [hunk 1]
```diff
@@ -103,29 +102,41 @@ class	array_1d	: public boost::numeric::ublas::vector_expression< array_1d<T, N>
     array_1d ():
         vector_expression<self_type> ()
     {
-        //sin esto no funciona en windows!!
-        //std::fill (data().begin(), data().end(), value_type	());
+        std::fill(data().begin(), data().end(), value_type());
```
Comment (pooyan-dadvand) [inline kratos/containers/array_1d.h]: @KratosMultiphysics/technical-committee don't agree with this change.

We avoid this for performance reasons. If someone needs to initialize it he/she may use the ZeroVector or ScalarVector while constructing. For setting to zero one can also call the clear method.

File: kratos/containers/array_1d.h [hunk 2]
```diff
@@ -103,29 +102,41 @@ class	array_1d	: public boost::numeric::ublas::vector_expression< array_1d<T, N>
...
+
     explicit BOOST_UBLAS_INLINE
     array_1d (size_type array_size):
         vector_expression<self_type> ()
     {
-        //sin esto no funciona en windows!!
-        //std::fill (data().begin(), data().end(), value_type	());
-
-        /* 				std::fill (data().begin(), data().end(), value_type	()); */
+        std::fill(data().begin(), data().end(), value_type());
```
Comment (pooyan-dadvand) [inline kratos/containers/array_1d.h]: again

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/technical-committee agrees with adding the constructor with initialization list but avoiding the filling of the additional elements.

**Comment by philbucher** [comment]
@KratosMultiphysics/technical-committee now this PR is compliant with what we discussed

Now it basically adds the CTor with initializer list and some tests

**Comment by philbucher** [comment]
@KratosMultiphysics/technical-committee I applied the requested changes

File: kratos/tests/cpp_tests/containers/test_array_1d.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license:
+//                   kratos/license.txt
+//
+//  Main authors:    Philipp Bucher (https://github.com/philbucher)
```
Comment (rubenzorrilla) [inline kratos/tests/cpp_tests/containers/test_array_1d.cpp]:
```suggestion
//  License:         BSD License
//                   Kratos default license: kratos/license.txt
//
//  Main authors:    Philipp Bucher (https://github.com/philbucher)
```

**Review by rubenzorrilla — DISMISSED** [review]

**Review by rubenzorrilla — APPROVED** [review]

**Comment by philbucher** [comment]
FYI @frawahl I think you also asked for initializer_list support

**Pull request merged**